Assets/Scripts/Movement.cs

```diff
@@ -81,9 +98,124 @@ private void FixedUpdate()
             Velocity.y = -2f;
         }
 
-     
+        findWhipPoint();
+    }
+
+    private void ChargingTorch(InputAction.CallbackContext obj)
+    {
+        
+        print("Holding");
+        isTorchButtonHeldDown = true;
+       
+    }
+
+    private void WhileChargingTorchThrow()
+    {
+        if (isTorchButtonHeldDown)
+        {
+            Vector3 CurrentTorchPosition = HeldTorch.transform.localPosition;
+            HeldTorch.transform.localPosition = new Vector3(CurrentTorchPosition.x, CurrentTorchPosition.y - 0.1f, CurrentTorchPosition.z);
+        } else
+        {
+            HeldTorch.transform.localPosition = DefaultTorchPosition;
+        }
+    }
+
+    private void TossTorch(InputAction.CallbackContext obj)
+    {
+            print(obj.duration);
+
+            //Direction Correction
+            Vector3 forceDirection = camera.transform.forward;
+            RaycastHit hit;
+
+            if(Physics.Raycast(camera.transform.position, forceDirection, out hit, 500f))
+            {
+            forceDirection = (hit.point - TorchSpawnPoint.transform.position).normalized;
+            }
+
+            ForwardsForce += ForwardsForce * (float)obj.duration * TorchForceMultiplier;
+            GameObject TorchInstance = Instantiate(Torch, TorchSpawnPoint.transform.position, camera.transform.rotation);
+            Rigidbody torch_rb = TorchInstance.GetComponent<Rigidbody>();
+            torch_rb.velocity = this.Velocity;
+            Vector3 ForceToAdd = forceDirection * ForwardsForce + TorchSpawnPoint.transform.up * UpwardsForce;
+            torch_rb.AddForce(ForceToAdd, ForceMode.Impulse);
+            torch_rb.AddTorque(transform.up * Random.Range(0, 20));
+            print("Rock and Stone brother: " + ForwardsForce);
+            ForwardsForce = DefaultForwardsForce;
+            isTorchButtonHeldDown = false;
+    }
+
+    IEnumerator WhipCoolDownTimer()
+    {
+        yield return new WaitForSeconds(whipCoolDownTime);
+        canFireWhip = true;
+    }
+
+    private void findWhipPoint() 
+    {
+
```

Comment:
There's a mix of number of spaces used for indenting here, I'd prefer to stick to 4 spaces everywhere

Reply:
sorry